docs/source/library/python/mix_ins/index.rst

```diff
@@ -0,0 +1,12 @@
+:mod:`mix-ins` --- Mix-in Classes
+=================================
+
+The classes in these modules are mix-ins that combined with the base
+:py:class:`ale.base.base.Driver` class create the concrete drivers.
```

Comment:
I wanted to add a reference to the concrete drivers index here but I couldn't get it to work for some reason.

Reply:
Is this something you want to spend more time trying to find a solution for or just bench it for now?

Reply:
I'm going to bench it for right now. I think this PR has gotten large enough. The next priority for documentation is ensuring that all of the Python and C++ code has documentation.